src/store/ShareStore.js

```diff
@@ -29,52 +29,100 @@ export default class ShareStore {
 
 
     @observable shareLink = '';
+
+    fixedEncodeURIComponent(str) {
+        return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
+            return `%${c.charCodeAt(0).toString(16)}`;
+        });
+    }
     refereshShareLink = () => {
         let self = this;
         if(!this.context || !this.menu.dialog.open ) { return; }
 
         const layoutData = this.stx.exportLayout(true);
         layoutData.favorites = [];
 
-        const json = JSON.stringify(layoutData),
-            fixedEncodeURIComponent = function (str) {
-                return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
-                    return `%${c.charCodeAt(0).toString(16)}`;
-                });
-            };
+        const json = JSON.stringify(layoutData);
 
         const origin = window.location.href;
-        this.shareLink = fixedEncodeURIComponent(`${origin}#${json}`);
+        const encodedJson = encodeURIComponent(json); // this.fixedEncodeURIComponent(json);
+
+        const parts = json.match(/.{1,1800}/g);
+
 
         this.shortUrlFailed = false;
         this.loading = true;
+        this.shareLink = '';
 
-        fetch(`${this.bitlyUrl}?access_token=${this.accessToken}&longUrl=${this.shareLink}`)
-            .then( response => {
-                return response.json();
-            })
-            .then( response =>  {
-                if (response.status_code == 200 ) {
-                    self.shareLink = response.data.url;
+        let hashPromise = Promise.resolve('NONE');
+        parts.forEach(part => {
+            hashPromise = hashPromise.then(hash => this.shortenBitlyAsync(part, hash));
+        });
+
+        hashPromise
+            .then(hash => {
+                if (hash) {
+                    self.shareLink = `https://bit.ly/${hash}`;
                     self.urlGenerated = true;
-                }else{
+                } else {
                     self.shortUrlFailed = true;
                     self.urlGenerated = false;
                 }
                 self.loading = false;
+
             })
             .catch(error => {
                 self.loading = false;
                 self.urlGenerated = false;
             });
+    }
+    shortenBitlyAsync(payload, hash) {
+        let origin = window.location.href;
+        origin = origin.replace('localhost', '127.0.0.1'); // make it work on localhost
+
+        //const shareLink = this.fixedEncodeURIComponent(`${origin}?${hash}#${payload}`);
+        const shareLink = encodeURIComponent(`${origin}?${hash}#${payload}`);
+
+        return fetch(`${this.bitlyUrl}/shorten?access_token=${this.accessToken}&longUrl=${shareLink}`)
+            .then(response => response.json())
+            .then(response =>  {
+                if (response.status_code == 200 ) {
+                    return response.data.url.split('bit.ly/')[1];
+                }
+                return null;
+            });
+    }
+    expandBitlyAsync(hash, payload) {
```

Comment:
just a thought: this function is only used in `ChartStore.js`. Though... It seems more appropriate to place it in `ShareStore.js` even though it is not used there.